CHANGELOG.md

```diff
@@ -18,6 +18,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Changed
 
 ### Dependencies
+- Update Apache Lucene from 10.1.0 to 10.2.1 ([#17961](https://github.com/opensearch-project/OpenSearch/pull/17961))
```

Comment:
10.2.1 supports JDK24. https://github.com/apache/lucene/pull/14300

I wonder if we tried to compile with JDK24 and see if there's any performance improvement from JDK24 Vectorization?

Reply:
@yupeng9 Some JDK 24 testing has happened in #18085 but I don't believe we've tested it with Lucene 10.2.1 yet.

Reply:
Thanks for bringing it up @yupeng9 

I wonder if there is a flag we are missing while using Java 21 with similar to `-Dorg.apache.lucene.store.MMapDirectory.enableMemorySegments=false` in the past that might make the performance numbers better. 

Also, should we decouple JDK 24 from this PR ? We would want to see the improvements Lucene would bring AS IS and then introduce JDK 24 ?